### PR TITLE
feat: limit permissions granted to AppStudio users

### PR DESCRIPTION
**Pull request by MatousJobanek**
This PR changes the `appstudio` tier template:

cleanup of :ghost::
* drops `Role` & `RoleBindings` that granted view & exec permissions to crtadmin users - we don't use crtadmin users in AppStudio cluster
* drops `RoleBinding` that granted view permissions to a `User` resource - there is no user provisoned for `appstudio` tier - we should use only SAs

limit the user's permissions:
* changes the `ClusterRole` that is granted to the user's SA in the workspace namespace from `edit` to `view`
* adds new `Role` & `RoleBinding` that grants these permissions to the user's namespace:
  * CRUD on HAS Application & Component CRDs
  * CRUD on PipelineRuns
  * only create & delete permissions for secrets
  
I also added some basic comments describing the purpose of the Roles/RoleBindings

for more info see: https://coreos.slack.com/archives/C02CTEB3MMF/p1643044182095500

paired PR: https://github.com/codeready-toolchain/toolchain-e2e/pull/456

### CODE REVIEW

File: deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml [hunk 1]
```diff
@@ -14,93 +14,98 @@ objects:
...
+    - applications
+    - components
     verbs:
     - get
     - list
     - watch
     - create
     - delete
     - update
+    - patch
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml:44]: Could be just * I guess.

**Review by alexeykazakov — APPROVED** [review]

**Review by rajivnathan — APPROVED** [review]
Looks good 👍

File: deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml (hunk 1 above)
Comment (MatousJobanek) [reply deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml]: good point - updated in bfe899bd

**Pull request merged**